traits/src/grid.rs

```diff
@@ -1,13 +1,15 @@
 //! Geometry and topology definitions
 
-pub use crate::cell::ReferenceCell;
-pub use crate::element::FiniteElement;
+use solvers_tools::arrays::AdjacencyList;
 
 pub trait Geometry {
-    fn reference_cell(&self) -> &dyn ReferenceCell;
-    fn map(&self, reference_coords: &[f64], physical_coords: &mut [f64]);
-
     fn dim(&self) -> usize;
+
+    fn point(&self, i: usize) -> Option<&[f64]>;
+
+    unsafe fn point_unchecked(&self, i: usize) -> &[f64];
+
+    fn point_count(&self) -> usize;
 }
 
 pub enum Locality {
```

Comment:
Why is this a pub enum and not a trait? Also, two aspects seem to be mixed here. One is to check the whether an element is local or global, the other one is to provide a map from local indices to global indices. Calling the second one `Locality` seems unusual. 

Should we maybe have a separate ```DofMap``` (or similarly named trait) to separate those two things?

Reply:
Going to remove this for now and just have this pull request for a serial grid then will work in parallel grid in a separate pr